### DISCUSSION TITLE
Mobile View

### DISCUSSION THREAD

**Discussion by Ampurcell** [cortexrd/Knack-Toolkit-Library#178] [General]
Hello Again, 

I feel like someone must have brought this up before but I couldn't find much. I am interested in adapting the views or pages based on a user's device. 

Knack recently updated to show previews in the mobile and tablet views which is a great help! My team and I use the app on the computer and mobile but in the past, I created pages only for mobile view and they would need to navigate to that page. 
I think it would be much more efficient if I could "hide" views based on the device. Or possibly hide columns, etc based on the device type. 

Do you have any documentation on something like that or similar? 

OH! also trying to figure out how to disable bulk delete for specific views.

**Comment by cortexrd** [comment]
Hi Alisha,

About hiding elements based on device type:  Currently, there's no keyword for that.  But adding one would be easy.  Actually, it would part of the **ktlCond** option.

Like this:  `_hv=ktlCond, is, ktlMobile`

Will let you know when it's ready.

About disabling the bulk operations for a view, there's a keyword that I simply forgot to document.

`_nbo`

You can add parameters to specify which operation(s) you want to disable.
See this post:  https://github.com/cortexrd/Knack-Toolkit-Library/issues/21#issuecomment-1540661073
You can even apply this for certain role only, like all other keywords.

I will update the documentation shortly.

Cheers,
Norm

**Comment by Ampurcell** [comment]
nbo is perfect! Thank you! 

And that would be great if you could add a new keyword!

**Comment by Ampurcell** [comment]
:( .... the ktlMobile is not working for me. I think I am on the latest release. 
<img width="1552" alt="Screenshot 2024-01-08 at 3 34 09 PM" src="https://github.com/cortexrd/Knack-Toolkit-Library/assets/141195753/1dbed44f-cbe4-42c3-8073-5c9e0419eeb8">

**Comment by Ampurcell** [comment]
I thought I added them correctly. But the web page on Safari(PC), Chrome(PC), & Safari(Mobile) are all blank. 
![Screenshot 2024-01-08 at 3 37 22 PM](https://github.com/cortexrd/Knack-Toolkit-Library/assets/141195753/910b4683-8092-4d01-b391-088b8d460fe4)
![Screenshot 2024-01-08 at 3 37 32 PM](https://github.com/cortexrd/Knack-Toolkit-Library/assets/141195753/54447b7b-3a8c-498d-bdef-5b71a3a47fd6)

![Screenshot 2024-01-08 at 3 37 39 PM](https://github.com/cortexrd/Knack-Toolkit-Library/assets/141195753/d4845af8-143e-4d89-a548-2a0f22585556)

**Comment by cortexrd** [comment]
Your keywords are ok.

It seems that you are running a very old version: **KTL v0.13.14** from your screen grab.  See just below the blue highlight.  Also, there are many errors that have probably been fixed since then.

You should see the actual version in the **Version Info Bar** at top-right, if you have it enabled.  If you don't, you can enable it for the Developer role only, which is very useful and recommended.

This is what you should see:

![image](https://github.com/cortexrd/Knack-Toolkit-Library/assets/42851359/fee5eb71-3875-47ac-b890-e0ce1d4e29a5)

What is in your KTL loader?

You should see this, with blank quotes:

![image](https://github.com/cortexrd/Knack-Toolkit-Library/assets/42851359/da2761e1-25d1-4090-86a9-dc4bbe71bb8c)

Also, you may want to try refreshing your page with **Command+F5** to bypass the cache.

Norm
